src/writeData/subscriptions/components/StringParsingForm.tsx

```diff
@@ -0,0 +1,193 @@
+// Libraries
+import React, {FC, useState, useEffect} from 'react'
+
+// Components
+import {
+  Input,
+  Grid,
+  Form,
+  InputType,
+  Dropdown,
+  IconFont,
+  Icon,
+  Heading,
+  HeadingElement,
+  FontWeight,
+  AlignItems,
+  ComponentSize,
+  FlexDirection,
+  FlexBox,
+} from '@influxdata/clockface'
+import StringPatternInput from 'src/writeData/subscriptions/components/StringPatternInput'
+
+// Types
+import {Subscription} from 'src/types/subscriptions'
+
+// Utils
+import {handleValidation} from 'src/writeData/subscriptions/utils/form'
+
+// Styles
+import 'src/writeData/subscriptions/components/StringParsingForm.scss'
+
+interface Props {
+  formContent: Subscription
+  updateForm: (any) => void
+}
+
+const StringParsingForm: FC<Props> = ({formContent, updateForm}) => {
+  const ruleList = ['field', 'tag']
+  const [rule, setRule] = useState('')
+  useEffect(() => {
+    if (rule === 'field') {
+      formContent.stringFields = [
+        ...formContent.stringFields,
+        {
+          name: '',
+          pattern: '',
+        },
```

Comment:
similar to a previous comment - can we put this default state into a constant so it's clear that it is being overridden by formContent?